kotlin-result-coroutines/src/jvmTest/kotlin/com/github/michaelbull/result/coroutines/binding/AsyncSuspendableBindingTest.kt

```diff
@@ -144,11 +145,17 @@ class AsyncSuspendableBindingTest {
             return Err(BindingError.BindingErrorB)
         }
 
+        val dispatcherA = TestCoroutineDispatcher()
+        val dispatcherB = TestCoroutineDispatcher()
+        val dispatcherC = TestCoroutineDispatcher()
+
         runBlocking {
             val result = binding<Unit, BindingError> {
-                launch(newThread("Thread A")) { provideX().bind() }
-                launch(newThread("Thread B")) { provideY().bind() }
-                launch(newThread("Thread C")) { provideZ().bind() }
+                launch(dispatcherA) { provideX().bind() }
+                dispatcherA.advanceTimeBy(20)
+                launch(dispatcherB) { provideY().bind() }
+                dispatcherB.advanceTimeBy(20)
```

Comment:
perhaps a bit contrived 😅  trying to represent the two threads running to 20 milliseconds at approx the same time, though in reality dispatcherA here only needs to be fast forwarded to 2